docs/validation-rules/index.rst

```diff
@@ -224,6 +225,82 @@ UInt64
 - ``IsPositive()`` - checks whether ``ulong|ulong?`` property is positive **(greater than zero)**.
 - ``Required()`` - checks whether ``ulong?`` property is not ``null``.
 
+Validating nested objects
+=========================
+Sometimes there could be need to validate nested objects. You can do this by creating your own ``RulesProvider``. Let's say we have the following model:
+
+.. sourcecode:: csharp
+
+    public class OrderModel
+    {
+        public AddressModel Address { get; set; }
+    }
+
+    public class AddressModel
+    {
+        public string City { get; set; }
+        public string Street { get; set; }
+    }
+
+We'd like to be sure that ``City`` and ``Street`` in ``OrderModel`` nested object are not empty. To do that we have to create provider:
+
+.. sourcecode:: csharp
+
+    class AddressModelRulesProvider : IValitRulesProvider<AddressModel>
+    {
+        public IEnumerable<IValitRule<AddressModel>> GetRules()
+            => ValitRules<AddressModel>
+            	.Create()
+            	.Ensure(m => m.City, _=>_
+            		.Required())
+            	.Ensure(m => m.Street, _=>_
+            		.Required())
+            	.GetAllRules();
+    }
+
+Then we can validate our ``OrderModel`` using created provider:
```

Comment:
Same as above.
_Then we can validate our ``OrderModel`` by using the created provider:_